central/delegatedregistryconfig/datastore/datastore.go

```diff
@@ -0,0 +1,56 @@
+package datastore
+
+import (
+	"context"
+
+	"github.com/stackrox/rox/central/delegatedregistryconfig/store"
+	"github.com/stackrox/rox/central/role/resources"
+	"github.com/stackrox/rox/generated/storage"
+	"github.com/stackrox/rox/pkg/sac"
+)
+
+// DataStore is the entry point for modifying Delegated Registry Config data.
+//
+//go:generate mockgen-wrapper
+type DataStore interface {
+	GetConfig(context.Context) (*storage.DelegatedRegistryConfig, error)
+	UpsertConfig(context.Context, *storage.DelegatedRegistryConfig) error
+}
+
+// New returns an instance of DataStore.
+func New(store store.Store) DataStore {
+	return &datastoreImpl{
+		store: store,
+	}
+}
+
+var (
+	administrationSAC = sac.ForResource(resources.Administration)
+)
+
+type datastoreImpl struct {
+	store store.Store
+}
+
+// GetConfig returns Central's delegated registry config
+func (d *datastoreImpl) GetConfig(ctx context.Context) (*storage.DelegatedRegistryConfig, error) {
```

Comment:
In general I'm a fan of passing back the bool and not using nil as the flag that the value didn't exist